rts/lra/lra-filters/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -259,8 +259,12 @@ public void filter(ContainerRequestContext containerRequestContext) throws IOExc
         // store state with the current thread. TODO for the async version use containerRequestContext.setProperty("lra", Current.peek());
         Current.updateLRAContext(lraId, headers); // make the current LRA available to the called method
 
-        if (newLRA != null)
+        if (newLRA != null) {
```

Comment:
newLRA is always null in this case because of:
 if (isLongRunning)
            newLRA = null;

Reply:
Yes but in the incoming request filter on line 270 there is a call that sets the notion of current lraClient.setCurrentLRA(lraId); 

Then in the response filter there on line 366 the call Current.updateLRAContext(responseContext.getHeaders());
will puts the notion of current into the response headers:

```java
    public static void updateLRAContext(MultivaluedMap<String, Object> headers) {
        URL lraId = Current.peek();

        if (lraId != null)
            headers.putSingle(LRA_HTTP_HEADER, lraId);
        else
            headers.remove(LRA_HTTP_HEADER);
    }
```